src/Core/Settings.h

```diff
@@ -351,6 +351,7 @@ class IColumn;
     M(UInt64, partial_merge_join_left_table_buffer_bytes, 0, "If not 0 group left table blocks in bigger ones for left-side table in partial merge join. It uses up to 2x of specified memory per joining thread.", 0) \
     M(UInt64, partial_merge_join_rows_in_right_blocks, 65536, "Split right-hand joining data in blocks of specified size. It's a portion of data indexed by min-max values and possibly unloaded on disk.", 0) \
     M(UInt64, join_on_disk_max_files_to_merge, 64, "For MergeJoin on disk set how much files it's allowed to sort simultaneously. Then this value bigger then more memory used and then less disk I/O needed. Minimum is 2.", 0) \
+    M(Bool, enable_parallel_hash_join, false, "Enable paralle join algorithm.", 0)\
```

Comment:
Basic question: What are the reasons for this to be disabled by default? Seems like a great addition to have it always on.

Also, there is a typo: `paralle` -> `parallel`

Reply:
It maybe have unexpected behaviors, we want to keep it in experiement status

Reply:
done